{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\Hiury G\\Documents\\Git-Projeto\\TheWeatherApp\\.venv\\lib\\site-packages\\pymongo\\common.py:857: UserWarning: Unknown option ssl_cert_reqs\n",
      "  warnings.warn(str(exc))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pinged your deployment. You successfully connected to MongoDB!\n",
      "6506491a1835bb6b3b242595\n"
     ]
    }
   ],
   "source": [
    "from pymongo.mongo_client import MongoClient\n",
    "from pymongo.server_api import ServerApi\n",
    "from bson.objectid import ObjectId\n",
    "uri = \"mongodb+srv://hiury:XiDZ4mhYjS7j38aO@weatherdb.0ywfw7t.mongodb.net/WeatherDB?retryWrites=true&w=majority&ssl=true&ssl_cert_reqs=CERT_NONE\"\n",
    "# Create a new client and connect to the server\n",
    "client = MongoClient(uri)\n",
    "# Send a ping to confirm a successful connection\n",
    "try:\n",
    "    client.admin.command('ping')\n",
    "    print(\"Pinged your deployment. You successfully connected to MongoDB!\")\n",
    "    \n",
    "    db = client.WheatherDB\n",
    "    \n",
    "    conta = db.regiao\n",
    "    \n",
    "    _id = ObjectId()\n",
    "    \n",
    "    results = conta.insert_one({\"_id\": _id, \"teste\": \"Teste\",\"Dados_Metereologicos\": {\n",
    "                  \"$ref\" : \"Dados_Metereologicos\",\n",
    "                  \"$id\" : _id,\n",
    "                  \"$db\" : \"WheatherDB\",\n",
    "                  \"extraField\" : \"anything\"\n",
    "               } })\n",
    "    print(results.inserted_id)\n",
    "    client.close()\n",
    "except Exception as e:\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\Hiury G\\Documents\\Git-Projeto\\TheWeatherApp\\.venv\\lib\\site-packages\\pymongo\\common.py:857: UserWarning: Unknown option ssl_cert_reqs\n",
      "  warnings.warn(str(exc))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pinged your deployment. You successfully connected to MongoDB!\n",
      "{'_id': ObjectId('64ebf8abf5433dcbc4bb9d41'), 'regiao': 'Unipam', 'dados_metereologicos': [{'temperatura': 27, 'umidade': 12.12, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}, {'temperatura': 22, 'umidade': 80.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "{'_id': ObjectId('64ebfc564971aa9801b916f3'), 'regiao': 'Tiros', 'dados_metereologicos': [{'temperatura': 18, 'umidade': 90.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "{'_id': ObjectId('64efd79542f2e68d774a2323'), 'regiao': 'Centro', 'dados_metereologicos': [{'temperatura': 23, 'umidade': 90.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "{'_id': ObjectId('64efd9232d5b4ecb2b0b9d09'), 'regiao': 'Unipam2', 'dados_metereologicos': [{'temperatura': 23, 'umidade': 99.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "{'_id': ObjectId('64efdac83b5f010630315ab6'), 'regiao': 'Unipam3', 'dados_metereologicos': [{'temperatura': 32, 'umidade': 98.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "{'_id': ObjectId('64efdaf6a66d03d056a4d31e'), 'regiao': 'Unipam3', 'dados_metereologicos': [{'temperatura': 32, 'umidade': 98.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "{'_id': ObjectId('64f543fd80caf9cc95f75984'), 'regiao': 'Sul', 'dados_metereologicos': [{'temperatura': 121, 'umidade': 23.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "{'_id': ObjectId('64f5440480caf9cc95f75985'), 'regiao': 'Sul2', 'dados_metereologicos': [{'temperatura': 121, 'umidade': 23.0, 'dt_criacao': datetime.datetime(2023, 9, 8, 15, 11, 16, 841000)}]}\n",
      "<pymongo.cursor.Cursor object at 0x0000027118356D40>\n"
     ]
    }
   ],
   "source": [
    "from pymongo.mongo_client import MongoClient\n",
    "from pymongo.server_api import ServerApi\n",
    "from datetime import datetime\n",
    "uri = \"mongodb+srv://hiury:XiDZ4mhYjS7j38aO@weatherdb.0ywfw7t.mongodb.net/WeatherDB?retryWrites=true&w=majority&ssl=true&ssl_cert_reqs=CERT_NONE\"\n",
    "# Create a new client and connect to the server\n",
    "client = MongoClient(uri)\n",
    "# Send a ping to confirm a successful connection\n",
    "try:\n",
    "    client.admin.command('ping')\n",
    "    print(\"Pinged your deployment. You successfully connected to MongoDB!\")\n",
    "    \n",
    "    db = client.WheatherDB\n",
    "    \n",
    "    conta = db.regiao\n",
    "    \n",
    "    results = conta.find()\n",
    "    for result in results:\n",
    "        print(result)\n",
    "        \n",
    "    print(results)\n",
    "    client.close()\n",
    "except Exception as e:\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\Hiury G\\Documents\\Git-Projeto\\TheWeatherApp\\.venv\\lib\\site-packages\\pymongo\\common.py:857: UserWarning: Unknown option ssl_cert_reqs\n",
      "  warnings.warn(str(exc))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pinged your deployment. You successfully connected to MongoDB!\n",
      "{'_id': ObjectId('64fb5e6c65ebbb8f51d8c540'), 'teste': 'foda', 'listaEmbeded': [{'Funciona': 'Sim', 'fakeNatty': True, 'suco': True}, {'Funciona': 'Sim', 'fakeNatty': True, 'suco': True}, {'Funciona': 'Sim', 'fakeNatty': True, 'suco': True}, {'Funciona': 'Sim', 'fakeNatty': True, 'suco': True}], 'Funciona': ['True pra krlh']}\n",
      "None\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from pymongo.mongo_client import MongoClient\n",
    "from pymongo.server_api import ServerApi\n",
    "from datetime import datetime\n",
    "from dotenv import load_dotenv\n",
    "\n",
    "load_dotenv()\n",
    "MONGODB_URI = os.environ['MONGODB_URI']\n",
    "# Create a new client and connect to the server\n",
    "client = MongoClient(uri)\n",
    "# Send a ping to confirm a successful connection\n",
    "try:\n",
    "    client.admin.command('ping')\n",
    "    print(\"Pinged your deployment. You successfully connected to MongoDB!\")\n",
    "    \n",
    "    db = client.Teste\n",
    "    \n",
    "    conta = db.account\n",
    "    \n",
    "    results = conta.find_one({'listaEmbeded':{\"$exists\":True}})\n",
    "    print(results)\n",
    "    resultsUpdate = conta.update_many({'listaEmbeded':{\"$exists\":True}}, {\"$push\": {'listaEmbeded': {\"Funciona\": 'True pra krlh'}}})\n",
    "    results = conta.find_one({'dados_metereologicos':{\"$exists\":True}})\n",
    "    print(results)\n",
    "    client.close()\n",
    "except Exception as e:\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed to insert document. Status code: 201\n",
      "Response: {\"insertedIds\":[\"64ea12d45ec784dd9ea0d01d\"]}\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "url = \"https://sa-east-1.aws.data.mongodb-api.com/app/data-ulzvz/endpoint/data/v1/action/insertMany\"\n",
    "\n",
    "headers = {\n",
    "    \"Content-Type\": \"application/ejson\",\n",
    "    \"Accept\": \"application/json\",\n",
    "    \"apiKey\": \"jtg3PBsJB8tkQkkzToXEh4nYsU2ggBUfIDkqR1f73p4ijpiau288lLrGY8XYHUMF\"\n",
    "}\n",
    "\n",
    "data = {\n",
    "    \"dataSource\": \"WeatherDB\",\n",
    "    \"database\": \"mongodbVSCodePlaygroundDB\",\n",
    "    \"collection\": \"sales\",\n",
    "    \"documents\": [{\n",
    "        \"text\": \"Hello, world!\"\n",
    "    }]\n",
    "}\n",
    "\n",
    "response = requests.post(url, headers=headers, json=data)\n",
    "\n",
    "if response.status_code == 200:\n",
    "    print(\"Document inserted successfully!\")\n",
    "else:\n",
    "    print(f\"Failed to insert document. Status code: {response.status_code}\")\n",
    "    print(\"Response:\", response.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'DadosMetereologicos' object has no attribute 'update'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[5], line 14\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[39mfor\u001b[39;00m regiao \u001b[39min\u001b[39;00m Regiao\u001b[39m.\u001b[39mobjects:\n\u001b[0;32m     13\u001b[0m     \u001b[39mfor\u001b[39;00m dm \u001b[39min\u001b[39;00m regiao\u001b[39m.\u001b[39mdados_metereologicos:\n\u001b[1;32m---> 14\u001b[0m         dm\u001b[39m.\u001b[39;49mupdate(dt_criacao \u001b[39m=\u001b[39m datetime\u001b[39m.\u001b[39mdatetime\u001b[39m.\u001b[39mnow)\n\u001b[0;32m     15\u001b[0m     regiao\u001b[39m.\u001b[39msave()\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'DadosMetereologicos' object has no attribute 'update'"
     ]
    }
   ],
   "source": [
    "from theweather.models import Regiao\n",
    "from weatherapp import settings\n",
    "from django.utils import timezone\n",
    "from django.conf import settings\n",
    "import datetime\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# Iterate through existing documents and update the dt_criacao field\n",
    "\n",
    "for regiao in Regiao.objects:\n",
    "    for dm in regiao.dados_metereologicos:\n",
    "        print(dm)\n",
    "    regiao.save()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
